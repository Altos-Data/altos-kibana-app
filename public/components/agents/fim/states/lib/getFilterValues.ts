/*
 * Wazuh app - Integrity monitoring components
 * Copyright (C) 2015-2020 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */
import { WzRequest } from '../../../../../react-services/wz-request';

export async function getFilterValues(field, value) {

  const filter = {
    distinct: true,
    select: field,
    limit: 30,
  };
  if (value) {
    filter['search'] = value;
  }
  // TODO: Get the agent from paramentes
  const result = await WzRequest.apiReq('GET', '/syscheck/001', filter)
  return (((result || {}).data || {}).data || {}).items.map((item) => {return item[field]});
}