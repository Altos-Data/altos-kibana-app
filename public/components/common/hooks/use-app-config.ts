
import { AppRootState } from '../../../redux/types';
import { useSelector } from 'react-redux';

export const useAppConfig = () => {
  const appConfig = useSelector((state: AppRootState) => state.appConfigState);
  return appConfig;
}
