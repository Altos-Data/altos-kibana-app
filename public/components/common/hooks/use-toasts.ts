/*
 * Wazuh app - React hook for get Kibana notifications toast service
 * Copyright (C) 2015-2020 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */

import { AppDependencies } from '../../../types';
import { useKibana } from '../../../../../../src/plugins/kibana_react/public';
import { ToastsStart } from 'kibana/public';

export const useToasts = (): ToastsStart => {
  const { core } = useKibana().services as AppDependencies;
  return core.notifications.toasts;
};
