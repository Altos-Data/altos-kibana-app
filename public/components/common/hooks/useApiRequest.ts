import { useState, useEffect } from 'react';
import WzRequest from '../../../react-services/wz-request';
import { WzApiResponse } from './types';




export function useApiRequest(method, path, params): [boolean, WzApiResponse, (string | undefined)] {
  const [items, setItems] = useState<WzApiResponse>({ affected_items: [], failed_items: [], total_affected_items: 0, total_failed_items: 0 });
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState<string | undefined>();
  useEffect(() => {
    setLoading(true);
    setError(undefined);
    WzRequest.apiReq(method, path, { params })
      .then(response => {
        setItems(response.data.data);
        setLoading(false);
      })
      .catch(error => {
        setError(error);
        setLoading(false)
      })
  }, [path, params]);
  return [loading, items, error];
}
