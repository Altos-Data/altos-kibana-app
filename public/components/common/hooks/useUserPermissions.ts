/*
 * Wazuh app - React hook for get query of Kibana searchBar
 * Copyright (C) 2015-2020 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */

import { useSelector } from 'react-redux';
import { checkMissingUserPermissions } from '../../../react-services/rbac';

// It retuns user permissions
export const useUserPermissions = () => {
  const userPermissions = useSelector(state => state.appStateReducers.userPermissions);
  return userPermissions;
}

// It returns user permissions validation and user permissions
export const useUserPermissionsValidation = (requiredPermissions) => {
  const userPermissions = useUserPermissions();
  const requiredPermissionsArray = typeof requiredPermissions === 'function' ? requiredPermissions() : requiredPermissions;
  return [checkMissingUserPermissions(requiredPermissionsArray, userPermissions), userPermissions];
}

// It redirects to other URL if user permissions are not valid
export const useUserPermissionsPrivate = (requiredPermissions, redirectURL) => {
  const [userPermissionsValidation, userPermissions] = useUserPermissionsValidation(requiredPermissions);
  if(userPermissionsValidation){
    window.location.href = redirectURL;
  }
  return [userPermissionsValidation, userPermissions];
}