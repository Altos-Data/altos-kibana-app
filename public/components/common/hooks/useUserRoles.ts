/*
 * Wazuh app - React hooks to manage user role requirements
 * Copyright (C) 2015-2020 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */

import { useSelector } from 'react-redux';
import { checkMissingUserPermissions } from '../../../react-services/rbac';

// It retuns user permissions
export const useUserRoles = () => {
  const userPermissions = useSelector(state => state.appStateReducers.userRoles);
  return userPermissions;
}

// It returns user permissions validation and user permissions
export const useUserRolesRequirements = (requiredRoles) => {
  const userRoles = useUserRoles();
  if(requiredRoles === null){
    return [false, userRoles]
  }
  const requiredPermissionsArray = typeof requiredRoles === 'function' ? requiredRoles() : requiredRoles;
  const rolesUserNotOwn = requiredPermissionsArray.filter(role => !userRoles.includes(role));
  return [rolesUserNotOwn.length ? rolesUserNotOwn : false, userRoles];
}

// It redirects to other URL if user permissions are not valid
export const useUserRolesPrivate = (requiredRoles, redirectURL) => {
  const [userPermissionsValidation, userPermissions] = useUserRolesRequirements(requiredRoles);
  if(userPermissionsValidation){
    window.location.href = redirectURL;
  }
  return [userPermissionsValidation, userPermissions];
}