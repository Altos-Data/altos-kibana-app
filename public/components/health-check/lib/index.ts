export { checkKibanaSettings } from './check-kibana-settings';
export { checkKibanaSettingsTimeFilter } from './check-time-filter';