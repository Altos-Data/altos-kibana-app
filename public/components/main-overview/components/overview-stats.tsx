/*
 * Wazuh app - React component for showing stats about agents.
 *
 * Copyright (C) 2015-2020 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */
import React from 'react';
import { EuiStat, EuiFlexItem, EuiFlexGroup, EuiPage, EuiToolTip } from '@elastic/eui';
import { AgentsSummary } from '../../common/hooks/types';

interface IOverviewStastsProps {
  summary: AgentsSummary;
}

export const OverviewStats = (props: IOverviewStastsProps) => {
  const goToAgents = (status) => {
    if (status) {
      sessionStorage.setItem(
        'agents_preview_selected_options',
        JSON.stringify([{ field: 'q', value: `status=${status}` }])
      );
    } else if (sessionStorage.getItem('agents_preview_selected_options')) {
      sessionStorage.removeItem('agents_preview_selected_options');
    }
    window.location.href = '#/agents-preview';
  };

  return (
    <EuiPage>
      <EuiFlexGroup>
        <EuiFlexItem />
        <EuiFlexItem>
          <EuiStat
            title={
              <EuiToolTip position="top" content={`Go to all agents`}>
                <span
                  className={'statWithLink'}
                  style={{ cursor: 'pointer' }}
                  onClick={() => goToAgents(null)}                  
                  data-test-subj='spanTotalAgents'
                >
                  {props.summary.total || '-'}
                </span>
              </EuiToolTip>
            }
            description="Total agents"
            titleColor="primary"
            textAlign="center"
          />
        </EuiFlexItem>
        <EuiFlexItem>
          <EuiStat
            title={
              <EuiToolTip position="top" content={`Go to active agents`}>
                <span
                  onClick={() => goToAgents('active')}
                  className={'statWithLink'}
                  style={{ cursor: 'pointer' }}
                  data-test-subj='spanActiveAgents'
                >
                  {props.summary.active || '-'}
                </span>
              </EuiToolTip>
            }
            description="Active agents"
            titleColor="secondary"
            textAlign="center"
          />
        </EuiFlexItem>
        <EuiFlexItem>
          <EuiStat
            title={
              <EuiToolTip position="top" content={`Go to disconnected agents`}>
                <span
                  onClick={() => goToAgents('disconnected')}
                  className={'statWithLink'}
                  style={{ cursor: 'pointer' }}
                  data-test-subj='spanDisconnectedAgents'
                >
                  {props.summary.disconnected || '-'}
                </span>
              </EuiToolTip>
            }
            description="Disconnected agents"
            titleColor="danger"
            textAlign="center"
          />
        </EuiFlexItem>
        <EuiFlexItem>
          <EuiStat
            title={
              <EuiToolTip position="top" content={`Go to never connected agents`}>
                <span
                  onClick={() => goToAgents('never_connected')}
                  className={'statWithLink'}
                  style={{ cursor: 'pointer' }}
                  data-test-subj='spanNeverConnectedAgents'
                >
                  {props.summary.never_connected || '-'}
                </span>
              </EuiToolTip>
            }
            description="Never connected agents"
            titleColor="subdued"
            textAlign="center"
          />
        </EuiFlexItem>
        <EuiFlexItem />
      </EuiFlexGroup>
    </EuiPage>
  );
};
