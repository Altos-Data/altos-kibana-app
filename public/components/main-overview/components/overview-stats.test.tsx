/*
 * Wazuh app - React component for main overview - Test
 *
 * Copyright (C) 2015-2020 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */

import React from 'react';
import { mount, shallow } from 'enzyme';
import { OverviewStats } from './overview-stats';
import { AgentsSummary } from '../../common/hooks/types';
import { findTestSubject } from '@elastic/eui/lib/test';

describe('Overview stats', () => {
  test('should render a Overview stats screen', () => {
    const summaryMock: AgentsSummary = {
      total: 0,
      active: 0,
      disconnected: 0,
      never_connected: 0,
    };
    const component = shallow(<OverviewStats summary={summaryMock} />);

    expect(component).toMatchSnapshot();
  });

  test('should render agents values with -', () => {
    const summaryMock: AgentsSummary = {
      total: 0,
      active: 0,
      disconnected: 0,
      never_connected: 0,
    };
    const component = mount(<OverviewStats summary={summaryMock} />);
    const totalAgents = findTestSubject(component, 'spanTotalAgents');
    expect(totalAgents.text()).toBe('-');
    const activeAgents = findTestSubject(component, 'spanActiveAgents');
    expect(activeAgents.text()).toBe('-');
    const disconnectedAgents = findTestSubject(component, 'spanDisconnectedAgents');
    expect(disconnectedAgents.text()).toBe('-');
    const neverConnectedAgents = findTestSubject(component, 'spanNeverConnectedAgents');
    expect(neverConnectedAgents.text()).toBe('-');
  });

  test('should render agents values with correct values', () => {
    const summaryMock: AgentsSummary = {
      total: 6,
      active: 3,
      disconnected: 2,
      never_connected: 1,
    };
    const component = mount(<OverviewStats summary={summaryMock} />);
    const totalAgents = findTestSubject(component, 'spanTotalAgents');
    expect(totalAgents.text()).toBe('6');
    const activeAgents = findTestSubject(component, 'spanActiveAgents');
    expect(activeAgents.text()).toBe('3');
    const disconnectedAgents = findTestSubject(component, 'spanDisconnectedAgents');
    expect(disconnectedAgents.text()).toBe('2');
    const neverConnectedAgents = findTestSubject(component, 'spanNeverConnectedAgents');
    expect(neverConnectedAgents.text()).toBe('1');
  });
});
