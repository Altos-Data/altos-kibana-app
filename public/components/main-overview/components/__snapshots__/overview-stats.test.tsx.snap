// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview stats should render a Overview stats screen 1`] = `
<EuiPage>
  <EuiFlexGroup>
    <EuiFlexItem />
    <EuiFlexItem>
      <EuiStat
        description="Total agents"
        textAlign="center"
        title={
          <EuiToolTip
            content="Go to all agents"
            delay="regular"
            position="top"
          >
            <span
              className="statWithLink"
              data-test-subj="spanTotalAgents"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              -
            </span>
          </EuiToolTip>
        }
        titleColor="primary"
      />
    </EuiFlexItem>
    <EuiFlexItem>
      <EuiStat
        description="Active agents"
        textAlign="center"
        title={
          <EuiToolTip
            content="Go to active agents"
            delay="regular"
            position="top"
          >
            <span
              className="statWithLink"
              data-test-subj="spanActiveAgents"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              -
            </span>
          </EuiToolTip>
        }
        titleColor="secondary"
      />
    </EuiFlexItem>
    <EuiFlexItem>
      <EuiStat
        description="Disconnected agents"
        textAlign="center"
        title={
          <EuiToolTip
            content="Go to disconnected agents"
            delay="regular"
            position="top"
          >
            <span
              className="statWithLink"
              data-test-subj="spanDisconnectedAgents"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              -
            </span>
          </EuiToolTip>
        }
        titleColor="danger"
      />
    </EuiFlexItem>
    <EuiFlexItem>
      <EuiStat
        description="Never connected agents"
        textAlign="center"
        title={
          <EuiToolTip
            content="Go to never connected agents"
            delay="regular"
            position="top"
          >
            <span
              className="statWithLink"
              data-test-subj="spanNeverConnectedAgents"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              -
            </span>
          </EuiToolTip>
        }
        titleColor="subdued"
      />
    </EuiFlexItem>
    <EuiFlexItem />
  </EuiFlexGroup>
</EuiPage>
`;
